src/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeBuilder.cs

```diff
@@ -190,7 +190,8 @@ public override void WriteString(string text)
 
         public override void WriteSurrogateCharEntity(char lowCh, char highCh)
         {
-            AddString(new string(new char[] { highCh, lowCh }));
+            ReadOnlySpan<char> entity = stackalloc char[] { highCh, lowCh };
```

Comment:
It's odd that some of these are ordered `highCh, lowCh` and some are ordered `lowCh, highCh`.  Do we know why?  Regardless, I'm not changing it as part of this PR, but it caught my eye.

Reply:
looks like a bug to me almost sure `hi, low` is correct. Surrogates are rarely used so I suspect very little people have used it or they didn't notice (or perhaps encoder fallback hid the problem or some other error elsewhere negates this). Do you feel like digging deeper? if not I think we should file an issue to take a closer look

Reply:
I'll open an issue for it.